be/src/formats/parquet/file_writer.cpp

```diff
@@ -177,139 +192,718 @@ void ParquetBuildHelper::build_compression_type(::parquet::WriterProperties::Bui
     }
 }
 
+std::shared_ptr<::parquet::schema::GroupNode> ParquetBuildHelper::make_schema(
+        const std::vector<std::string>& file_column_names, const std::vector<ExprContext*>& output_expr_ctxs) {
+    ::parquet::schema::NodeVector fields;
+
+    for (int i = 0; i < output_expr_ctxs.size(); i++) {
+        auto column_expr = output_expr_ctxs[i]->root();
+        auto nodePtr = _make_schema_node(
+                file_column_names[i], column_expr->type(),
+                column_expr->is_nullable() ? ::parquet::Repetition::OPTIONAL : ::parquet::Repetition::REQUIRED);
+        DCHECK(nodePtr != nullptr);
+        fields.push_back(std::move(nodePtr));
+    }
+
+    return std::static_pointer_cast<::parquet::schema::GroupNode>(
+            ::parquet::schema::GroupNode::Make("table", ::parquet::Repetition::REQUIRED, fields));
+}
+
+std::shared_ptr<::parquet::WriterProperties> ParquetBuildHelper::make_properties(const ParquetBuilderOptions& options) {
+    ::parquet::WriterProperties::Builder builder;
+    builder.version(::parquet::ParquetVersion::PARQUET_2_0);
+    options.use_dict ? builder.enable_dictionary() : builder.disable_dictionary();
+    starrocks::parquet::ParquetBuildHelper::build_compression_type(builder, options.compression_type);
+    return builder.build();
+}
+
+// Repetition of subtype in nested type is set by default now, due to type descriptor has no nullable field.
```

Comment:
@letian-jiang @stephen-shelby According to Iceberg spec, the sub field of nested type can be non-nullable.

Reply:
Yes. However, `TypeDescriptor` currently does not have a field to denote nullability. Thus, we have to set it by default as of now.

Reply:
This may cause inconsistent schemas in the Iceberg table sink. What do you think? @stephen-shelby